controllers/ocirepository_controller.go

```diff
@@ -1125,3 +1143,29 @@ func (r *OCIRepositoryReconciler) notify(ctx context.Context, oldObj, newObj *so
 		}
 	}
 }
+
+// ociSourceConfigChanged evaluates the current spec with the observations
+// of the artifact in the status to determine if source configuration has
+// changed and requires rebuilding the artifact.
+func ociSourceConfigChanged(obj *sourcev1.OCIRepository) bool {
+	if !pointer.StringEqual(obj.Spec.Ignore, obj.Status.ObservedIgnore) {
+		return true
+	}
+
+	if !layerSelectorEqual(obj.Spec.LayerSelector, obj.Status.ObservedLayerSelector) {
+		return true
+	}
+
+	return false
+}
+
+// Based on k8s.io/utils/pointer/pointer.go pointer value equality.
+func layerSelectorEqual(a, b *sourcev1.OCILayerSelector) bool {
```

Comment:
Maybe?
```suggestion
// Returns true if both arguments are nil or both arguments
// dereference to the same value.
func layerSelectorEqual(a, b *sourcev1.OCILayerSelector) bool {
```

Reply:
No longer applicable as this has been removed now.